src/services/item/plugins/html/service.ts

```diff
@@ -0,0 +1,196 @@
+
+  /**
+   * Helper to build the local or remote path of the package file
+   * // <contentId>/<filename>.<extension>
+   */
+  buildPackagePath = (rootPath: string, filename: string) =>
+    path.join(rootPath, `${filename}.${this.extension}`);
+
+  /**
+   * Helper to build the local or remote path of the html content root
+   * // <contentId>/content
+   */
+  buildContentPath = (rootPath: string) => path.join(rootPath, 'content');
+
+  /**
+   * Uploads the package content into storage
+   * Recursive function to traverse and upload the html folder
+   * IMPORTANT: the top-down traversal must not wait for long (ensures that files
+   * are uploaded in parallel as soon as possible). Results aggregation can however
+   * await in parallel (note: await in a map fn does not block the map iteration).
+   */
+  async upload(
+    member: Actor,
+    folder: string,
+    uploadPath: string,
+    log?: FastifyBaseLogger,
+  ): Promise<Array<string>> {
+    if (!member) {
+      throw new UnauthorizedMember(member);
+    }
+
+    const children = await readdir(folder);
+
+    // we will flatMap with promises: first map
+    const uploads = children.map(async (child) => {
+      const childPath = path.join(folder, child);
+      const childUploadPath = path.join(uploadPath, child);
+      const stats = await lstat(childPath);
+
+      if (stats.isDirectory()) {
+        // recursively upload child folder
+        return await this.upload(member, childPath, childUploadPath, log);
+      } else {
+        // ignore this file if extension is not allowed
+        const ext = path.extname(childPath);
+
+        if (!this.validator.isExtensionAllowed(ext)) {
+          log?.info('HTML import: illegal file extension detected, skipping file: ', ext);
+          // we're using flatMap, represent none value with empty array
+          return [];
+        }
+
+        const mimetype = mime.getType(ext) ?? DEFAULT_MIME_TYPE;
+        await this.fileService.upload(member, {
+          file: fs.createReadStream(childPath),
+          filepath: childUploadPath,
+          mimetype,
+        });
+
+        // we're using flatMap, wrap result in array
+        return [childUploadPath];
+      }
+    });
+    // then resolve promises array and flatten
+    return (await Promise.all(uploads)).flat();
+  }
+
+  async createItem(
+    actor: Member,
+    repositories: Repositories,
+    htmlFile: MultipartFile,
+    onComplete: (
+      actor: Member,
+      baseName: string,
+      contentId: string,
+      parentId?: Item['id'],
+    ) => Promise<Item>,
+    parentId?: Item['id'],
+    log?: FastifyBaseLogger,
+  ): Promise<Item> {
+    /*
+      // uppy tries to guess the mime type but fails and falls back to application/octet-stream, so we disable this check for now
+      if (htmlFile.mimetype !== H5P_FILE_MIME_TYPE) {
+        throw new InvalidH5PFileError(htmlFile.mimetype);
+      }
+      */
```

Comment:
Remove, or provide a way against the default mime type (could be application/octet-stream just to be safe).

Reply:
I don't know it was disabled in the first place? Should I enable it back?

Reply:
Not sure about the behvaiour of uppy: does it always provide the `application/octet-stream` mime type? I guess it's a minor validation anyway, in the sense that it would help us eliminate requests that attempt to upload unwanted filetypes, but it does not really validate if the file really matches the mimetype anyway.